cyclops-ui/src/components/pages/Modules/Modules.tsx

```diff
@@ -88,108 +125,207 @@ const Modules = () => {
+                  style={{
+                    whiteSpace: "nowrap",
+                    overflow: "hidden",
+                    textOverflow: "ellipsis",
+                    display: "block",
+                  }}
+                >
+                  Repo:
+                  <Link aria-level={3} href={module.template.repo}>
+                    {" " + module.template.repo}
+                  </Link>
+                </Col>
+              </Row>
+              <Row gutter={[16, 16]}>
+                <Col
+                  span={24}
+                  style={{
+                    whiteSpace: "nowrap",
+                    overflow: "hidden",
+                    textOverflow: "ellipsis",
+                    display: "block",
+                  }}
+                >
+                  Path:
+                  <Link
+                    aria-level={3}
+                    href={
+                      module.template.repo +
+                      `/tree/` +
+                      getTemplateVersion(
+                        module.template.version,
+                        module.template.resolvedVersion,
+                      ) +
+                      `/` +
+                      module.template.path
+                    }
+                  >
+                    {" " + module.template.path}
+                  </Link>
+                </Col>
+              </Row>
+              <Row gutter={[16, 16]}>
+                <Col
+                  span={24}
+                  style={{
+                    whiteSpace: "nowrap",
+                    overflow: "hidden",
+                    textOverflow: "ellipsis",
+                    display: "block",
+                  }}
+                >
+                  Version:
+                  <span style={{ color: "#1677ff" }}>
+                    {" "}
+                    {getTemplateVersion(
                       module.template.version,
                       module.template.resolvedVersion,
-                    ) +
-                    `/` +
-                    module.template.path
-                  }
-                >
-                  {" " + module.template.path}
-                </Link>
-              </Col>
-            </Row>
-            <Row gutter={[16, 16]}>
-              <Col
-                span={24}
-                style={{
-                  whiteSpace: "nowrap",
-                  overflow: "hidden",
-                  textOverflow: "ellipsis",
-                  display: "block",
-                }}
-              >
-                Version:
-                <span style={{ color: "#1677ff" }}>
-                  {" "}
-                  {getTemplateVersion(
-                    module.template.version,
-                    module.template.resolvedVersion,
+                    )}
+                  </span>
+                </Col>
+              </Row>
+            </Card>
+          </a>
+        </Col>
+      ));
+    }
+    return filteredData
```

Comment:
Can we do the filtering when checkboxes change and save the filtered modules into the `filteredData`? That way, you don't have to change the part that represents the modules.

Check `handleSearch` function from the same file

Reply:
This is handled during checking or returning the the value

Reply:
This should trigger when checkboxes change